rust/libceed/src/vector.rs

```diff
@@ -269,6 +269,21 @@ impl<'a> Vector<'a> {
         Ok(ptr_copy)
     }
 
+    /// Copy the array of self into vec_copy
+    ///
+    /// # arguments
+    ///
+    /// * `vec_copy` - vector to copy array values to
+    #[allow(unused_mut)]
+    fn vec_copy_mut(
+        mut self,
+        vec_copy: crate::Vector,
+    ) -> crate::Result<bind_ceed::CeedVector> {
+        let ierr = unsafe { bind_ceed::CeedVectorCopy(self, vec_copy) };
+        self.check_error(ierr)?;
+        Ok(vec_copy)
+    }
```

Comment:
Hmm, this looks like a situation where you may be fighting the borrow checker. I think this would prevent some ownership headaches:
```suggestion
    fn copy_from(
        mut self,
        vec_source: &crate::Vector,
    ) -> crate::Result<bind_ceed::CeedVector> {
        let ierr = unsafe { bind_ceed::CeedVectorCopy(vec_source.ptr, self.ptr) };
        self.check_error(ierr)?;
        Ok(self)
    }
```